AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -207,7 +207,14 @@ - (void)removeAuthToken:(nullable NSString *)authToken {
   if (history) {
     MSLogDebug([MSAppCenter logTag], @"Retrieved history state.");
   } else {
-    MSLogWarning([MSAppCenter logTag], @"Failed to retrieve history state or none was found.");
+    if (decryptedData) {
+      MSLogWarning([MSAppCenter logTag], @"Failed to unarchive decryptedData.");
+    } else if (encryptedData) {
```

Comment:
Isn't it `!encryptedData`?
Please check this condition before the line 205.

Reply:
no. when code comes here. means history is null and we need to log warning if decrpytedData is NOT null, meaning [NSKeyedUnarchiver unarchiveObjectWithData:decryptedData] is not returning correctly.

Reply:
we should not simply check null and log warning by each step. because people can pass in null. we should log warning only when previous steps are not returning null but current step is returning null.

Reply:
but we will try to think it in a more intuitive way